guides/creating-a-custom-stepper-using-the-cdk-stepper.md

```diff
@@ -0,0 +1,153 @@
+# Creating a custom stepper using the CDK stepper
+
+Using the [CDK stepper](https://material.angular.io/cdk/stepper/overview) it is possible to build a custom stepper which you can completely style yourself without any specific Material Design styling.
+
+In this guide, we'll learn how we can build our own custom stepper using the CDK stepper.
+
+## Add CDK stepper to your project 
+
+Add the CDK to your Angular project using npm or yarn:
+
+__npm__
+```bash
+npm install --save @angular/cdk
+```
+
+__yarn__
+```bash
+yarn add @angular/cdk
+```
+
+Next step is to add the `CdkStepperModule` to your Angular module:
+
+```typescript
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { CdkStepperModule } from '@angular/cdk/stepper'; // this is the relevant import
+
+import { AppComponent } from './app.component';
+
+@NgModule({
+  imports: [BrowserModule, CdkStepperModule], // add the module to your imports
+  declarations: [AppComponent],
+  bootstrap: [AppComponent],
+})
+export class AppModule {}
+```
+
+## Create our custom stepper component
+
+Now we are ready to create our custom stepper component. Therefore, we need to create a new Angular component which extends from `CdkStepper`:
+
+__custom-stepper.component.ts__
+```typescript
+import { Directionality } from '@angular/cdk/bidi';
+import { ChangeDetectorRef, Component, Input, QueryList } from '@angular/core';
+import { CdkStep, CdkStepper } from '@angular/cdk/stepper';
+
+@Component({
+  selector: 'app-custom-stepper',
+  templateUrl: './custom-stepper.component.html',
+  styleUrls: ['./custom-stepper.component.css'],
+  providers: [{ provide: CdkStepper, useExisting: CustomStepperComponent }],
+})
+export class CustomStepperComponent extends CdkStepper {
+  /** Whether the validity of previous steps should be checked or not */
+  linear: boolean;
+
+  /** The index of the selected step. */
+  selectedIndex: number;
+
+  /** The list of step components that the stepper is holding. */
+  _steps: QueryList<CdkStep>;
+
+  constructor(dir: Directionality, changeDetectorRef: ChangeDetectorRef) {
+    super(dir, changeDetectorRef);
+  }
+}
+```
+
+As we extend our component class from `CdkStepper` we can now access different properties from this class like `linear` and `selectedIndex` which are defined in the [API documentation](https://material.angular.io/cdk/stepper/api#CdkStepper). Additionally, we can also access non-documented properties like `_steps` which we need in our example to be able to iterate over all available steps:
```

Comment:
"As we extend" -> "After we've extended". Also maybe it makes sense to document/expose some of these undocumented properties, if they're require to implement your own stepper?

Reply:
Adjusted the text.

I would also agree that it woul make sense to make steps a public getter but I think this was made private by purpose?!

Reply:
Can we rename it in this PR to just `steps`. I don't want our official documentation to tell people to use underscored properties, they'll get into a bad habit. It is an internal property so we can probably just get away with renaming it, but if we want to be safe we can alias the old property with a getter/setter

Reply:
@mmalerba : Ok, I renamed it to `steps`.

Reply:
Done